### PR TITLE
@broskoski => Stand-alone signup and login page

### PR DESCRIPTION
**Pull request by mzikherman**
This adds in the client-side handling for the email/password form.

Fb/twitter just works cause of AP(!)

### CODE REVIEW

File: desktop/apps/auth/client/auth.coffee [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return unless @validateForm()
+    return if @formIsSubmitting()
+
+    e.preventDefault()
+
+    action = $(e.target).data('action')
+    @afterAuthHref = $(e.target).data('after-auth')
+
+    @user.set (data = @serializeForm())
+    @user[action]
```
Comment (mzikherman) [inline desktop/apps/auth/client/auth.coffee]: We can almost use the same little form view here for both signup and login. Use the template to keep which 'action' it is.

**Comment by mzikherman** [comment]
screen shot 2017-03-15 at 1 24 15 pm

You get the usual behavior ^

File: desktop/apps/auth/client/auth.coffee (hunk 1 above)
Comment (broskoski) [inline desktop/apps/auth/client/auth.coffee]: Nice

File: desktop/assets/auth.coffee [hunk 2]
```diff
@@ -0,0 +1,9 @@
+require('backbone').$ = $
+
+routes =
+  '/signup': require('../apps/auth/client/auth.coffee').init
+
+  '/login': require('../apps/auth/client/auth.coffee').init
+
+for path, init of routes
+  $(init) if location.pathname.match path
```
Comment (broskoski) [inline desktop/assets/auth.coffee]: Just curious, when would the auth client not initialize? Is it more simple just to `init` always?

File: desktop/components/main_layout/footer/view.coffee [hunk 3]
```diff
@@ -28,7 +27,7 @@ module.exports = class FooterView extends Backbone.View
 
   show: ->
     @$el.show()
-    reInitRVARail(@$recentlyViewedArtworks) if shouldShowRVARail()
+    reInitRVARail(@$recentlyViewedArtworks) if shouldShowRVARail() && @$recentlyViewedArtworks.length > 0
```
Comment (broskoski) [inline desktop/components/main_layout/footer/view.coffee]: Hello

**Comment by broskoski** [comment]
LGTM, the auth client is just the kind of backbone view we like to see here at Artsy HQ. A+

File: desktop/components/main_layout/footer/view.coffee (hunk 3 above)
Comment (mzikherman) [inline desktop/components/main_layout/footer/view.coffee]: I noticed this crash b/c it's initialized as part of the main layout asset package. But in the case of the blank main layout template, this element isn't here. /shrug

File: desktop/assets/auth.coffee (hunk 2 above)
Comment (mzikherman) [inline desktop/assets/auth.coffee]: Oh yea good call, since it's in its own asset package 👍

**Comment by mzikherman** [comment]
> LGTM, the auth client is just the kind of backbone view we like to see here at Artsy HQ. A+

brb re-writing in typescript and react

**Pull request merged**